fluent-fallback/src/types.rs

```diff
@@ -53,7 +53,7 @@ pub enum ResourceType {
 }
 
 /// A resource identifier for a localization resource.
-#[derive(Debug, Clone, Hash)]
```

Comment:
Question (blocking): Which clippy rule is this? I don't understand this change (probably since I don't know the clippy rule)

Reply:
https://rust-lang.github.io/rust-clippy/master/index.html#derive_hash_xor_eq

There's an `impl PartialEq` which only uses `self.value` for equality testing, while the derived `Hash` would use both `value` & `resource_type`, which breaks the invariant `k1 == k2 => hash(k1) == hash(k2)` (from the clippy rule)